modules/twinkleblock.js

```diff
@@ -433,9 +433,37 @@ Twinkle.block.callback.change_action = function twinkleblockCallbackChangeAction
 			tags: true,
 			placeholder: 'Enter pages to block user from',
 			maximumSelectionLength: 10, // Software limitation [[phab:T202776]]
-			language: {
-				noResults: function() {
-					return 'No pages entered yet';
+			minimumInputLength: 1, // prevent ajax call when empty
+			createTag: function() { // prevent selection of arbitrary titles not returned by API
+				return null;
+			},
+			ajax: {
+				url: mw.util.wikiScript('api'),
+				dataType: 'json',
+				data: function(params) {
+					if (!params.term) {
```

Comment:
Actually, it *would* be good to check that it's not empty; `mw.Title` trims whitespace, but gets cranky if there's *only* whitespace.  I don't think there's a visual issue here, but logs an error.

```suggestion
					if (!params.term.trim()) {
```
Or something like it.

Reply:
Actually actually, it might be be better to just verify that `mw.Title` was successful below, as it also throws an error when you've got just the namespace entered (e.g. "User:").  Again, nothing visual in the module AFAICT, but a console error.

Reply:
> Is this necessary? When does select2 ever return an empty params?

When you first click on it, when there's nothing in the search field (I'd've expected the library to supply `params.term = ''` in such a case, but it doesn't). But ah, I did add `minimumInputLength: 1` to handle this better (doesn't give "Search results could not be loaded" text), so yeah this check is unnecessary now. You're right again 👌

Reply:
The trouble with `mw.Title` is that it throws. We've to use `mw.Title.newFromText` (which is exception-safe) to prevent console errors.